hw/ip/sram_ctrl/data/sram_ctrl.hjson

```diff
@@ -0,0 +1,178 @@
+  // Interrupts and Alerts //
+  ///////////////////////////
+
+  interrupt_list: [
+    { name: "sram_parity_error",
+      desc: "A uncorrectable parity error has occurred."
+    }
+  ],
+
+  ////////////////
+  // Parameters //
+  ////////////////
+  param_list: [
+
+  ]
+
+  /////////////////////////////
+  // Intermodule Connections //
+  /////////////////////////////
+
+  inter_signal_list: [
+    // Key request to OTP
+    { struct:  "sram_otp_key"
+      type:    "req_rsp"
+      name:    "sram_otp_key"
+      act:     "req"
+      default: "'0"
+      package: "otp_ctrl_pkg"
+    }
+  ] // inter_signal_list
+
+  regwidth: "32",
+  registers: [
+
+    ////////////////////////
+    // Ctrl / Status CSRs //
+    ////////////////////////
+
+    { name: "STATUS",
+      desc: "SRAM status register.",
+      swaccess: "rw1c",
+      hwaccess: "hrw",
+      hwqe:     "true",
+      hwext:    "true",
+      fields: [
+        { bits: "0"
+          name: "ERROR"
+          desc: '''
+                Set to 1 if an uncorrectable parity error has occurred.
+                If set to 1, SW should check the !!ERROR_ADDRESS register for more information.
+                Writing 1 to this register clears the error bit.
+                '''
+        }
+        { bits: "1"
+          name: "SCR_KEY_VALID"
+          desc: '''
+                Set to 1 if a new scrambling key has been successfully obtained from OTP.
+                Note that if this is set to 0, the SRAM contents are still scrambled, but a
+                default all-zero key and nonce are used to do so.
+                '''
+        }
+        { bits: "2"
+          name: "SCR_KEY_SEED_VALID"
+          desc: '''
+                Set to 1 if the scrambling key has been derived from a valid key seed in OTP.
+                If !!STATUS.SCR_KEY_VALID is set to 1, !!STATUS.SCR_KEY_SEED_VALID should be 1
+                except for cases where the scrambling key seeds have not yet been provisioned to
+                OTP. In such a case, the scrambling key is still ephemeral (i.e., it is derived
+                using entropy from CSRNG), but a default all-zero value is used as the key seed.
+                '''
+        }
+      ]
+    }
+    { name: "CTRL_REGWEN",
+      desc: "Lock register for control register.",
+      swaccess: "rw1c",
+      hwaccess: "none",
+      fields: [
+        { bits: "0"
+          desc: '''
+                When cleared to zero, !!CTRL can not be written anymore.
+                ''',
+          resval: 1
+        }
+      ]
+    }
+    { name: "CTRL",
+      desc: "SRAM ctrl register.",
+      swaccess: "r0w1c",
+      hwaccess: "hro",
+      hwqe:     "true",
+      hwext:    "false",
+      regwen:   "CTRL_REGWEN"
+      fields: [
+        { bits: "0",
+          name: "RENEW_SCR_KEY",
+          desc: '''
+                Write 1 to request a new scrambling key from OTP. After writing to this register, software must not
```

Comment:
might be good to mention what would happen if during the process of renewal, this bit is written to again.
Probably nothing?

Reply:
nothing will happen in that case. I'll add that to the description.